File: Pemprograman_6/Bahasa_C/PRAK604-2510817120005-NorKhalisyah.c
#include <stdio.h>
#include <string.h>

int main() {
    char kode[1000], pesan[1000];

    printf("Masukkan kode Shikamaru: ");
    if (fgets(kode, sizeof(kode), stdin) == NULL) return 1;
    kode[strcspn(kode, "\n")] = 0;

    printf("Masukkan pesan yang diterima: ");
    if (fgets(pesan, sizeof(pesan), stdin) == NULL) return 1;
    pesan[strcspn(pesan, "\n")] = 0;

    int panjang_kode = strlen(kode);
    int panjang_pesan = strlen(pesan);

    if (panjang_kode != panjang_pesan) {
        printf("Panjang kalimat berbeda, pesan palsu\n");
        return 0;
    }

    char hasil[panjang_kode + 1];
    int jumlah_bintang = 0;
    int jumlah_pagar = 0;

    for (int i = 0; i < panjang_kode; i++) {
        if (kode[i] == pesan[i]) {
            hasil[i] = '*';
            jumlah_bintang++;
        } else {
            hasil[i] = '#';
            jumlah_pagar++;
        }
    }
    hasil[panjang_kode] = '\0';

    printf("%s\n", hasil);
    printf("*=%d\n", jumlah_bintang);
    printf("#=%d\n", jumlah_pagar);

    if (jumlah_bintang >= jumlah_pagar) {
        printf("Pesan Asli\n");
    } else {
        printf("Pesan Palsu\n");
    }

    return 0;
}

File: Pemprograman_6/Bahasa_Python/PRAK604-2510817120005-NorKhalisyah.py
kode = input("Masukkan kode Shikamaru: ").strip()
pesan = input("Masukkan pesan yang diterima: ").strip()

if len(kode) != len(pesan):
    print("Panjang kalimat berbeda, pesan palsu")
    exit()

hasil = []
jumlah_bintang = 0
jumlah_pagar = 0

for i in range(len(kode)):
    if kode[i] == pesan[i]:
        hasil.append('*')
        jumlah_bintang += 1
    else:
        hasil.append('#')
        jumlah_pagar += 1

print(''.join(hasil))
print(f"* = {jumlah_bintang}")
print(f"# = {jumlah_pagar}")

if jumlah_bintang >= jumlah_pagar:
    print("Pesan Asli")
else:
    print("Pesan Palsu")
